pkg/controller/certificates/sync.go

```diff
@@ -823,7 +823,9 @@ func setSecretValues(ctx context.Context, crt *cmapi.Certificate, s *corev1.Secr
 			return err
 		}
 
-		s.Annotations[cmapi.CommonNameAnnotationKey] = x509Cert.Subject.CommonName
+		if len(x509Cert.Subject.CommonName) > 0 {
+			s.Annotations[cmapi.CommonNameAnnotationKey] = x509Cert.Subject.CommonName
+		}
```

Comment:
Does it make sense to set this to `""` if it isn't set? This code will omit the annotation altogether if not set on the certificate.

Reply:
What's the advantage to have the key existing as an empty string?

Reply:
It just seems more consistent, and it's what we did previously 🙄

Reply:
Sgtm :+1: